src/main/java/ch/njol/skript/conditions/CondServerTickState.java

```diff
@@ -0,0 +1,95 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.conditions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.bukkitutil.ServerUtils;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.RequiredPlugins;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Condition;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import org.bukkit.Bukkit;
+import org.bukkit.ServerTickManager;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+import java.util.Locale;
+
+@Name("Server Tick State")
+@Description("Represents the ticking state of the server, for example, if the server is frozen, or running normally.")
+@Examples({
+	"if server's tick state is currently frozen:",
+	"if server tick state is normal:"
+})
+@Since("INSERT VERSION")
+@RequiredPlugins("Minecraft 1.20.4+")
+public class CondServerTickState extends Condition {
+
+	private ServerState state;
+
+	static {
+		if (ServerUtils.isServerTickManagerPresent())
+			Skript.registerCondition(CondServerTickState.class,
+				"[the] server['s] tick[ing] state is [currently] (:frozen|:stepping|:sprinting|:normal)",
+				"[the] server['s] tick[ing] state is(n't| not) [currently] (:frozen|:stepping|:sprinting|:normal)");
```

Comment:
I'm not a huge fan of this syntax. It makes it seem like `server tick state` should be an expression. Maybe we could have `server (ticking|tick[ing] state)`? It is a little weird for stepping/sprinting, so other syntax ideas are welcome.

Reply:
I'm also a fan of `tick rate`, or even just `if the server is stepping`, though that might be a bit ambiguous.

Reply:
> I'm not a huge fan of this syntax. It makes it seem like `server tick state` should be an expression. Maybe we could have `server (ticking|tick[ing] state)`? It is a little weird for stepping/sprinting, so other syntax ideas are welcome.

mmm, I can't think of any other syntax that would be a good fit here, and also sound good in your head, for your suggestion `if the server ticking is currently frozen` doesn't sound great, and the suggestion that Sovde put is good, but as he said, may be a bit ambiguous. I personally think the syntax is fine for now, unless we get a better suggestion

Reply:
Since the effects are `freeze server` and `make server sprint`, I think the ambiguity argument is a little silly actually. If it matches the effects, it should be fine.

Reply:
so `if the server is frozen` should be valid? I think thats fine.